cdap-watchdog/src/main/java/co/cask/cdap/logging/logbuffer/cleaner/LogBufferCleanerService.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.logbuffer.cleaner;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.logging.logbuffer.LogBufferFileOffset;
+import co.cask.cdap.logging.meta.CheckpointManager;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Log Buffer cleaner service. It periodically cleans up log buffer files for which logs are already persisted.
```

Comment:
I think the cleanup should be purely files size based, no need to be periodic. E.g. every time when a new WAL log file is created, it also check the total size of all the WAL files. It it goes over a limit, run an async job to cut the size down to some value coming from the configuration. I believe each WAL file size are about the same (variant due to the last event that makes the file goes over the max file size).

Reply:
Fixed. Now using `LogBufferCleaner` runnable which is passed from LogBufferService